libs/blocks/merch/merch.js

```diff
@@ -24,86 +18,97 @@ const GEO_MAPPINGS = {
   no: 'nb-NO',
 };
 
-const PLACEHOLDER_TYPE_MAPPINGS = {
-  optical: 'priceOptical',
-  strikethrough: 'priceStrikethrough',
-};
-
-let initialized = false;
-
-function omitNullValues(target) {
+const omitNullValues = (target) => {
   if (target != null) {
     Object.entries(target).forEach(([key, value]) => {
       if (value == null) delete target[key];
     });
   }
   return target;
-}
+};
 
-const getTacocatEnv = (envName, locale) => {
-  const wcsLocale = GEO_MAPPINGS[locale.prefix] ?? locale.ietf;
-  // eslint-disable-next-line prefer-const
-  let [language, country = 'US'] = wcsLocale.split('-', 2);
+window.tacocat = window.tacocat || {};
+
+export const imsCountryPromise = () => new Promise((resolve) => {
+  let count = 0;
+  const check = setInterval(() => {
+    count += 1;
+    if (window.adobeIMS) {
+      clearInterval(check);
+      if (window.adobeIMS.isSignedInUser()) {
+        window.adobeIMS
+          .getProfile()
+          .then(({ countryCode }) => resolve(countryCode))
+          .catch(() => resolve());
+      } else {
+        resolve();
+      }
+    } else if (count > 25) {
+      clearInterval(check);
+      resolve();
+    }
+  }, 200);
+});
+window.tacocat.imsCountryPromise = imsCountryPromise();
+
+export const getTacocatEnv = (envName, locale) => {
+  const wcsLocale = (GEO_MAPPINGS[locale.prefix] ?? locale.ietf).split('-', 2);
+  let language = wcsLocale[0];
+  const country = wcsLocale[1] || 'US';
   if (!SUPPORTED_LANGS.includes(language)) {
     language = 'en';
   }
   const host = envName === ENV_PROD
     ? 'https://www.adobe.com'
     : 'https://www.stage.adobe.com';
-  const scriptUrl = `${host}/special/tacocat/lib/${VERSION}/tacocat.js`;
+
   const literalScriptUrl = `${host}/special/tacocat/literals/${language}.js`;
-  return { scriptUrl, literalScriptUrl, country, language };
+  const scriptUrl = `${host}/special/tacocat/lib/${VERSION}/tacocat.js`;
+  const tacocatEnv = envName === ENV_PROD ? 'PRODUCTION' : 'STAGE';
+  return { literalScriptUrl, scriptUrl, country, language, tacocatEnv };
 };
 
-function initTacocat(envName, country, language) {
-  window.tacocat.tacocat({
-    defaults: {
-      country,
-      language,
-    },
-    environment: envName,
-    wcs: WCS,
-    literals: window.tacocat.literals?.[language] ?? {},
-  });
-}
-
-function loadTacocat() {
-  if (initialized) {
-    return;
-  }
-  initialized = true;
+window.tacocat.loadPromise = new Promise((resolve) => {
+  const { env, locale } = getConfig();
```

Comment:
we need landscape here, and pass it along. May be making env some object?

Reply:
we need to decide first where the landscape=draft would be stored, before passing it. so probably shouldn't be in scope of this pr

Reply:
IMO we should address landscape in a separate PR.